lib/WebService/SendBird.pm

```diff
@@ -0,0 +1,306 @@
+package WebService::SendBird;
+
+use strict;
+use warnings;
+
+use Mojo::UserAgent;
+use Mojo::URL;
+
+use WebService::SendBird::User;
+use WebService::SendBird::GroupChat;
+
+use Carp qw();
+
+our $VERSION = '0.001';
+
+# ABSTRACT: client to work with SendBird API
+
+=head1 NAME
+
+WebService::SendBird - unofficial support for the Sendbird API
+
+=head1 SYNOPSIS
+
+ use WebService::SendBird;
+
+ my $api = WebService::SendBird->new(
+     api_token => 'You_Api_Token_Here',
+     app_id    => 'You_App_ID_Here',
+ );
+
+ my $user = $api->create_user(
+     user_id     => 'my_chat_user_1',
+     nickname    => 'pumpkin',
+     profile_url => undef,
+ );
+
+ my $chat = $api->create_group_chat(
+     user_ids => [ $user->user_id ],
+ );
+
+=head1 DESCRIPTION
+
+Basic implementation for SendBird Platform API client, which helps to create users and group chats.
+
+More information at L<Platform API Documentation|https://docs.sendbird.com/platform>
+
+=cut
+
+use constant DEFAULT_API_URL_TEMPLATE => 'https://api-%s.sendbird.com/v3';
+use constant DEFAULT_REQUEST_TIMEOUT  => 15;
+
+=head1 METHODS
+
+=head2 new
+
+Creates an instance of API client
+
+=over 4
+
+=item * C<api_token> - Master or Secondary API Token.
+
+=item * C<app_id> - Application ID.
+
+=item * C<api_url> - URL to API end point. By default it will be generated from app_id.
+
+=item * C<ua> - Custom http client for API requests, should have the same interface like L<Mojo::UserAgent>.
+
+=item * C<timeout> - request timeout, default value 15 seconds
+
+=back
+
+=cut
+
+
+sub new {
+    my ($cls, %params) = @_;
+
+    Carp::croak('Missing required argument: api_token') unless $params{api_token};
+    Carp::croak('Missing required argument: app_id or api_url') unless $params{app_id} || $params{api_url};
```

Comment:
```suggestion
    Carp::croak('Missing required argument: app_id or api_url') unless $params{app_id} ^ $params{api_url};
```

Reply:
Why?

Reply:
It doesn't make sense to provide both, right?

Reply:
We need at least one of them, nothing bad in getting both. Error message will have no sense if we will throw it in case we get both arguments.